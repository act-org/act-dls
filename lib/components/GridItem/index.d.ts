/**
 * @prettier
 */
import * as React from 'react';
import { Props as GridBaseProps } from '~/components/GridBase';
export declare type Props = GridBaseProps;
declare const GridItem: React.FC<Props>;
export default GridItem;
//# sourceMappingURL=index.d.ts.map