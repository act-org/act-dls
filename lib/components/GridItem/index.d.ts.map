{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../src/components/GridItem/index.tsx"],"names":[],"mappings":"AAAA;;GAEG;AAEH,OAAO,KAAK,KAAK,MAAM,OAAO,CAAC;AAE/B,OAAiB,EAAE,KAAK,IAAI,aAAa,EAAE,MAAM,uBAAuB,CAAC;AAEzE,oBAAY,KAAK,GAAG,aAAa,CAAC;AAElC,QAAA,MAAM,QAAQ,EAAE,KAAK,CAAC,EAAE,CAAC,KAAK,CAE7B,CAAC;AAEF,eAAe,QAAQ,CAAC"}