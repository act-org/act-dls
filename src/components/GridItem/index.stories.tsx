/**
 * @prettier
 */

/* eslint-disable react/no-array-index-key, sort-keys */

import * as React from 'react';
import { select } from '@storybook/addon-knobs';

import GridContainer from '~/components/GridContainer';

import GridItem from '.';

export const item = (): React.ReactElement<any> => (
  <GridContainer
    spacing={select(
      'Spacing',
      {
        0: 0,
        1: 1,
        2: 2,
        3: 3,
        4: 4,
        5: 5,
        6: 6,
        7: 7,
        8: 8,
        9: 9,
        10: 10,
      },
      0,
    )}
  >
    {[
      ...new Array(
        select(
          'Quantity',
          {
            1: 1,
            2: 2,
            3: 3,
            4: 4,
            5: 5,
            6: 6,
            7: 7,
            8: 8,
            9: 9,
            10: 10,
          },
          1,
        ),
      ),
    ].map(
      (_, i): React.ReactElement<any> => (
        <GridItem key={i}>{`Grid Item ${i + 1}`}</GridItem>
      ),
    )}
  </GridContainer>
);

export default {
  component: GridItem,
  parameters: {
    info: {
      text: 'This is the item variant of the Grid component.',
    },
  },
  title: 'Atoms/Grid',
};
