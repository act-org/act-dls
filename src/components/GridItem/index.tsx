/**
 * @prettier
 */

import * as React from 'react';

import GridBase, { Props as GridBaseProps } from '~/components/GridBase';

export type Props = GridBaseProps;

const GridItem: React.FC<Props> = (props: Props): React.ReactElement<any> => (
  <GridBase item {...props} />
);

export default GridItem;
