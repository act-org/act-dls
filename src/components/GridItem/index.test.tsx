/**
 * @prettier
 */

import * as React from 'react';

import render from '~/helpers/test/render';

import GridItem from '.';

describe('GridItem', () => {
  it('matches the snapshot', () => {
    const { container } = render(<GridItem>GridItem</GridItem>);

    expect(container).toMatchSnapshot();
  });
});
