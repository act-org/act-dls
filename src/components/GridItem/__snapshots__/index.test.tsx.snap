// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridItem matches the snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-item"
  >
    GridItem
  </div>
</div>
`;
